pandas/tests/scalar/test_na_scalar.py

```diff
@@ -0,0 +1,108 @@
+import numpy as np
+import pytest
+
+from pandas.core.na_scalar import NA
+
+
+def test_singleton():
+    assert NA is NA
+    new_NA = type(NA)()
+    assert new_NA is NA
+
+
+def test_repr():
+    assert repr(NA) == "NA"
+    assert str(NA) == "NA"
+
+
+def test_truthiness():
+    with pytest.raises(TypeError):
+        bool(NA)
+
+    with pytest.raises(TypeError):
+        not NA
+
+
+def test_hashable():
+    assert hash(NA) == hash(NA)
+    d = {NA: "test"}
+    assert d[NA] == "test"
+
+
+def test_arithmetic_ops(all_arithmetic_functions):
```

Comment:
I think you should add tests with `np.nan` for both logical and arithmetic ops

Reply:
Good idea! added